SAML Authentication

MarkusHarmsen opened the pull request:
As discussed in #955, a SAML authentication could be useful for some users (at least it is for us).

Here is a proposal, on how to implement SAML authentication. It has been tested with the GSuite.

stevecrozz commented:
@MarkusHarmsen this looks great. I've been putting this PR off too long and I think its time to focus on it because there's a lot of value here for SAML users.

I don't particularly want the Vagrantfile, but still interested in why you have it. Are you just more familiar with Vagrant? or did you have a problem with the Docker setup?

MarkusHarmsen commented:
@stevecrozz I removed the Vagrant stuff and updated the pull request.

The reason for Vagrant was just that I'm more familiar with Vagrant than docker ;)

stevecrozz commented:
This is nice @MarkusHarmsen. One last question for you before I merge this. Are you willing to act as a sponsor for this feature, helping to answer questions and take occasional Errbit tickets relating to SAML auth? I ask because if we merge this, then we're also declaring that we support it. If our users start using this, they'll definitely raise issues and ask questions, and I don't always have the time to answer.

MarkusHarmsen commented:
Sure - before merging this, let me one time completely re-setup an errbit-instance an configure it using SAML (GSuite). I will document all required steps, so that might be a good starting point for a better documentation.

stevecrozz commented:
Sounds great @MarkusHarmsen!

stevecrozz commented:
@MarkusHarmsen it looks like there's another feature that will probably arrive in master soon that will conflict with this one: #1221. I don't think it will be very difficult to resolve, but it seems likely to happen.

stevecrozz commented:
@MarkusHarmsen This PR is so good and so close, I'd hate to let it fade away. Are you still intending to add more documentation?
